Removes lack of smoking-related stuff (actually)

TobyThorne opened the pull request:
Тревис, ты неиронично сука.

tl;dr перерабатываем курение и добавляем кучу связанного контента, добавлен табак, бумага для самокруток и новые вещества

- Добавлена мутация мака (poppy) - опиумный мак (opium poppy). Внезапно, содержит опиум.
- Добавлен опиум (opium). Являет собой усиленную версию трамадола, но при этом имеет более серьёзные побочные эффекты. Побочные эффекты проявляются при дозировке более 10 единиц, усиливаются при дозировке выше 15 единиц, передозировка наступает при 20 единицах. Включают в себя замедление, размытие зрения, сонливость.
- Добавлен тарин (tarine). Получается из опиума (3 opium, 1 acetone, 1 hydrochloric acid -> 3 tarine). Обезболивает сильнее оксикодона, но имеет очень серьёзные побочные эффекты. При любой дозе слегка замедляет, при дозировке выше 7.5 единиц замедляет ещё сильнее и вызывает галлюцинации, при передозировке (от 15 единиц) замедляет ещё сильнее, вызывает постоянные потери сознания и угнетает дыхание. 
- Побочные эффекты обоих веществ учитывают и переработанный, и находящийся непосредственно в крови объем. Следовательно, побочные эффекты могут наступить практически сразу после введения, а перед употреблением новой дозы безопаснее будет подождать, пока старая полностью выветрится из организма. Робастить героином вместо хлорки - новая ачивка, да.
- Добавлена бумага для самокруток, двух видов:
- Дешёвая GreySlide, белая.
- Дорогая RYR, неотбелённая и органическая. Курится дольше.
- Добавлены несколько видов табака для самокруток/трубок:
- StarLing Classic, обычный табак.
- StarLing Cherry, с вишней.
- StarLing Menthol, с ментолом.
- StarLing Chocolate, с шоколадом.
- Admiral's Pipe, высокого качества. Покупается за монетку.
- Rawdmann's, с анашой. Достаётся из взломанного автомата.
- Самокрутки крутятся кликом щепоткой табака по бумажке.
- Косяки теперь можно крутить не только из каннабиса, а из любых продуктов гидропоники, но требуется предварительное засушивание. Крутятся аналогично самокруткам. 
- Старым способом (с обычной бумагой) можно скрутить большой косяк (но не самокрутку).
- Исправлено потребление реагентов при курении. Раньше реагенты потреблялись с постоянной скоростью и могли полностью выкуриться незадолго после зажигания, после чего сигарета оставалась пустой. Теперь реагенты равномерно потребляются в течение всего процесса курения.
- Исправлено "активное" курение кликами по своему персонажу. Раньше такое курение было рудиментом - не потребляло реагенты и не было видно для окружающих. Теперь оно работает корректно и показывает окружающим аналогичные эмоутам сообщения. И имеет кулдаун, ибо нехер.
- Значительно увеличена доля реагентов, усваиваемых при курении. Зависит от вида - от 0.5 (против старых 0.2) у обычных сигарет с фильтрами до 0.9 у косяков.
- Добавлено появление дыма во время курения. Теперь можно ещё задорнее доводить медиков до истерик.
- Сокращено количество стартовых семян в хранилищах. Спрайт хранилища слегка изменён.
- Добавлены баки с компостом. В них можно кидать ненужные семена, перерабатывая их в компост.
- Добавлен компост. Являет собой слабое, но всё же достаточное для поддержания жизни растений удобрение.
- Усилен EZ Nutrient. Теперь все три "именных" удобрения имеют свои плюсы: EZ Nuntrient на 20% питательнее, Left4Zed увеличивает шанс мутаций на 20%, Robust Harvest увеличивает урожай на 20%. Для всех трёх нарисованы новые спрайты.
- Добавлен Mut'O'Grow - технический мутаген. Гораздо слабее "нормального", слегка токсичен для растений, но имеется у ботаников в автомате. 
- Удорожён рецепт трамадола. Рецепт тот же, но производит не 3, а 1 единицу трамадола.
- Улучшены спрайты сигарет, сигар, etc.
- Пофикшен огонёк от сигареты, появляющийся на предметах, на которых не должен.

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Commit:
oneToSquashThemAll

Тревис, ты неиронично сука.

TapKo4eB commented:
в пре мапдиффы приму после мапфиксов татара

Commit:
Merge branch 'dev' of https://github.com/ChaoticOnyx/OnyxBay into smokrework

Commit:
mapdate

TobyThorne commented:
Помержил карту

TapKo4eB commented on code/modules/reagents/reagent_dispenser.dm:
```diff
@@ -201,6 +201,25 @@
+	initial_reagent_types = list(/datum/reagent/toxin/fertilizer/compost = 1)
+	atom_flags = ATOM_FLAG_CLIMBABLE
+
+/obj/structure/reagent_dispensers/composttank/attackby(obj/item/W, mob/user)
+	src.add_fingerprint(user)
+	if(istype(W,/obj/item/seeds))
+		user.visible_message("[user] places [W] into \the [src].", \
+							 "You place [W] into \the [src].")
+		reagents.add_reagent(/datum/reagent/toxin/fertilizer/compost, 3)
+		qdel(W)
+	return ..()
```
а как же проверка на заполненность бачка

TobyThorne replied:
Бай дизайн. Чтобы можно было уничтожать семена без необходимости девать куда-то вёдра говна.

Commit:
Merge branch 'dev' of https://github.com/ChaoticOnyx/OnyxBay into smokrework

Epicus7 commented:
![image](https://user-images.githubusercontent.com/5204667/93545511-456ee800-f969-11ea-9af5-d69a20ef9047.png)
![image](https://user-images.githubusercontent.com/5204667/93545474-22dccf00-f969-11ea-9986-e1a6d3f6d9b1.png)

TapKo4eB committed:
pain

TapKo4eB committed:
pain2

TapKo4eB reviewed (dismissed):
нормально

TatarTheBuilder reviewed (dismissed):
отличный пр код не читал

Commit:
therewego

- Renamed RYR Papers to MAN Papers
- Fixed pipes' smoking time
- Reduced the amount of smoke produced by passive (holding a smokable in the mask slot) smoking;
- Fixed a typo;

Commit:
Merge branch 'smokrework' of https://github.com/TobyThorne/OnyxBay into smokrework

TobyThorne commented:
- RYR Papers переименованы в MAN Papers;
- Пофикшено время курения трубок, теперь они не будут прогорать за несколько секунд;
- Значительно уменьшено количества дыма при пассивном (сигарета в слоте маски) курении;
